### ISSUE TITLE
Uncaught TypeError: Converting circular structure to JSON

### ISSUE THREAD

**Issue by joggienl** [Countly/countly-sdk-web#79]
When posting a form, I got the following error: `Uncaught TypeError: Converting circular structure to JSON`

See the code at the following line https://github.com/Countly/countly-sdk-web/blob/da9e73278f3ce77d161eb702bbc711adea3fab5a/lib/countly.js#L1379

You cannot get the attributes from a form like that, see https://html.spec.whatwg.org/multipage/forms.html#dom-form-elements-dev for the defenition.

TL;DR
If you do `form[name]` (like `form.name` or `form.action`) you will get a form control that is using that `name` attribute. A small sample: https://codepen.io/joggie/pen/vYKNqYE

**Comment by ar2rsawseen** [comment]
Hmm, interestingly, but I cannot reproduce this :D
![image](https://user-images.githubusercontent.com/399992/97741082-e1ece400-1aea-11eb-9041-aa0aa66c530f.png)

But either way, you have a point, unfortunately, cannot use `getAttribute` because some clients need IE7 compatibility (some banks are still stuck with it). But will try to fall back to attributes property. 

Thank you for reporting :)

**Comment by ar2rsawseen** [comment]
Fixed here: https://github.com/Countly/countly-sdk-web/commit/96e77f04fd737ef5e40eada6db7e8f1d77ff2c23

**Comment by joggienl** [comment]
IE7!? Did not expect that to be needed 😅 

Are there plans to not have to support that anymore? I can imagine that could help the sdk as well.

As for the fix: nice! Now we wait for a new release :-)

Also weird you were not able to reproduce, did the codepen show the error?

**Comment by ar2rsawseen** [comment]
Hello,
sorry for the late reply, yes I later figured out the reason, I think it depended on how the form was defined, but don't remember the exact case currently.
But yes, we have such clients, so we need to be careful, what we put here :)
